driver/src/main/java/org/kaazing/k3po/driver/netty/channel/LocationFactories.java

```diff
@@ -38,14 +38,33 @@ public static LocationFactory changeSchemeOnly(String newScheme) {
     private static final class KeepAuthorityOnlyTransportFactory extends LocationFactory {
 
         private final String newScheme;
+        private String oldScheme;
 
-        public KeepAuthorityOnlyTransportFactory(String newScheme) {
+        public KeepAuthorityOnlyTransportFactory(String newScheme, String oldScheme) {
             this.newScheme = newScheme;
+            this.oldScheme = oldScheme;
         }
 
         @Override
         public URI createURI(URI location) {
-            return URI.create(format("%s://%s", newScheme, location.getAuthority()));
+            URI result = null;
+            switch (oldScheme) {
+            case "http":
+                if (location.getPort() == -1) {
+                    result = URI.create(format("%s://%s:80", newScheme, location.getAuthority()));
```

Comment:
maybe have a utility method to get default port for a URI, then you can combine the code into an if/else block based on location.port being -1, something like

if (location.getPort() == -1) {
    int port = getDefaultPortForScheme(location.getScheme());
    result = URI.create(format("%s://%s:%d", newScheme, location.getAuthority(), port));
} else {
    result = URI.create(format("%s://%s", newScheme, location.getAuthority()));
}